packages/realm-react/src/__tests__/useObjectRenderList.test.tsx

```diff
@@ -0,0 +1,319 @@
+    </SetupComponent>
+  );
+};
+
+const SetupComponent = ({ children }: { children: JSX.Element }): JSX.Element | null => {
+  const realm = useRealm();
+
+  useEffect(() => {
+    return () => {
+      realm.close();
+    };
+  }, [realm]);
+
+  const [setupComplete, setSetupComplete] = useState(false);
+  useEffect(() => {
+    realm.write(() => {
+      realm.deleteAll();
+      const items = testCollection.map((object) => realm.create(ListItem, object));
+      const list = realm.create(List, { id: 1, title: "List", items: items });
+      //testCollection.forEach((object) => list.items.push(realm.create(ListItem, object)));
+    });
+    setSetupComplete(true);
+  }, [realm]);
+
+  if (!setupComplete) {
+    return null;
+  }
+
+  return children;
+};
+
+const Item: React.FC<{ item: ListItem & Realm.Object }> = React.memo(({ item }) => {
+  renderCounter();
+  const realm = useRealm();
+  return (
+    <View testID={`result${item.id}`}>
+      <View testID={`name${item.id}`}>
+        <Text>{item.name}</Text>
+      </View>
+      <TextInput
+        testID={`input${item.id}`}
+        value={item.name}
+        onChangeText={(text) => {
+          realm.write(() => {
+            item.name = text;
+          });
+        }}
+      ></TextInput>
+      <TouchableHighlight
+        testID={`deleteButton${item.id}`}
+        onPress={() => {
+          realm.write(() => {
+            realm.delete(item);
+          });
+        }}
+      >
+        <Text>Delete</Text>
+      </TouchableHighlight>
+    </View>
+  );
+});
+
+const TestComponent = () => {
+  const list = useObject(List, 1);
+
+  const renderItem = useCallback(({ item }) => <Item item={item} />, []);
+
+  const keyExtractor = useCallback((item) => item.id, []);
+
+  return (
+    <>
+      <FlatList testID={"list"} data={list?.items ?? []} keyExtractor={keyExtractor} renderItem={renderItem} />;
+      {list?.favoriteItem && (
+        <View testID={`favoriteItem-${list?.favoriteItem.name}`}>
+          <Text>{list?.favoriteItem.name}</Text>
+        </View>
+      )}
+    </>
+  );
+};
+
+describe("useObjectRenderList: list property", () => {
+  afterEach(() => {
+    renderCounter.mockClear();
+    Realm.clearTestState();
+  });
+  afterAll(() => {
+    testRealm.close();
+  });
+
+  it("renders data in one render cycle per visible object in collection", async () => {
+    const { getByTestId } = render(<App />);
+
+    await waitFor(() => getByTestId("list"));
+
+    expect(renderCounter).toHaveBeenCalledTimes(10);
+  });
+  it("change to data will rerender", async () => {
+    const { getByTestId, getByText } = render(<App />);
+
+    const collection = testRealm.objectForPrimaryKey(List, 1)?.items ?? [];
+    const firstItem = collection[0];
+    const id = firstItem.id;
+
+    const nameElement = getByTestId(`name${id}`);
+    const input = getByTestId(`input${id}`);
+
+    expect(nameElement).toHaveTextContent(`${id}`);
+    expect(renderCounter).toHaveBeenCalledTimes(10);
```

Comment:
> It feels like it would make more sense to throw if objectForPrimaryKey returns null, as this should never happen, and the test will presumably fail on line 201 in this case anyway.

objectForPrimaryKey will return null if the primaryKey does not exist for said object, or if it's been deleted.  I don't think this should throw, as the developer should be adding null checks in their rendering code in case the object was suddenly deleted by sync.
That being said, there should be tests for this.